Memory bar for Mantid processes

sf1919 opened the pull request:
**Description of work.**
A second memory bar has been added to the widget showing how much memory Mantid processes are taking up. 

**To test:**
1. Open Mantid
2. Check that both memory bars are visible
3. Load a large file (e.g. a WISH file)
4. Check that both memory bars change

Review the code

Fixes #33619 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Add method for calculating process memory used

sf1919 committed:
Added second progress bar for Mantid memory usage

sf1919 committed:
Minor changes and release notes

- added release note for #33665
- Changed labels on bars
- ensured Mantid memory bar is the same colour as system memory bar

sf1919 committed:
Updated tests and release note

- added mantid memory bar to test mocking
- added tests for the new mantid memory bar
- added an image of the memory bar to release notes
- updated colour of mantid memory bar
- changed label on System memory bar

DavidFair commented on qt/python/mantidqt/mantidqt/widgets/memorywidget/memoryview.py:
```diff
@@ -71,9 +77,25 @@ def set_bar_color(self, current_value: int, new_value: int):
+            self.mantid_memory_bar.setStyleSheet(CRITICAL_STYLE)
+        elif from_critical_to_normal(self.critical, current_value, new_value):
+            self.mantid_memory_bar.setStyleSheet(NORMAL_STYLE)
+        else:
+            pass
```
Just omit the else branch in this case

DavidFair reviewed (changes requested):
Works for me, some slight improvements

sf1919 committed:
Added dataclasses and refreshed tests

- following review of #33665 moved the memory bar information into two dataclasses, one for system memory bar and one for the mantid memory bar
- Fixed tests and extended memoryinfo tests to check all attributes are of correct type and relevant sizes

sf1919 committed:
Remove unneccessary else branch in memoryview.py

DavidFair commented on qt/python/mantidqt/mantidqt/widgets/memorywidget/memoryinfo.py:
```diff
@@ -27,3 +29,22 @@ def get_memory_info() -> Tuple[int, float, float]:
     return mem_used_percent, mem_used_GB, mem_total_GB
+
+
+def get_mantid_memory_info() -> Tuple[int, float, float]:
```
It would be good to use the same struct, but we can do this as a follow up PR

DavidFair reviewed (approved):
LGTM and still works, lets :shipit:
